addons/l10n_fr_pdp_reports/models/pdp_flow.py

```diff
@@ -0,0 +1,752 @@
+        index=True,
+    )
+    batch_index = fields.Integer(
+        string="Batch Index",
+        default=1,
+        help="Allows splitting large reporting periods into multiple payloads.",
+    )
+    revision = fields.Integer(
+        string="Revision",
+        default=0,
+        help="Increments every time the payload is rebuilt for the same period.",
+        copy=False,
+    )
+    period_start = fields.Date(
+        string="Period Start",
+        help="First day covered by this PDP report.",
+        copy=False,
+    )
+    period_end = fields.Date(
+        string="Period End",
+        help="Last day covered by this PDP report.",
+        copy=False,
+    )
+    periodicity_code = fields.Char(
+        string="Periodicity Code",
+        help="Identifier of the reporting frequency provided by the PDP Annuaire.",
+        copy=False,
+    )
+    is_correction = fields.Boolean(
+        string="Correction Flow",
+        help="Set on supplemental flows generated after an initial payload was already sent.",
+        copy=False,
+    )
+    supplement_number = fields.Integer(
+        string="Supplement #",
+        default=0,
+        help="Incremented every time a new correction flow is generated for the same period.",
+        copy=False,
+    )
+    last_send_datetime = fields.Datetime(
+        string="Last Send On",
+        help="Datetime of the most recent send attempt to the PDP transport.",
+    )
+    send_datetime = fields.Datetime(
+        string="Sent On",
+        help="Datetime of the last successful submission to the PDP transport.",
+        copy=False,
+    )
+    attempt_count = fields.Integer(
+        string="Send Attempts",
+        default=0,
+        help="Number of times this flow was sent to the PDP transport.",
+    )
+    company_id = fields.Many2one(
+        "res.company",
+        string="Company",
+        required=True,
+        default=lambda self: self.env.company,
+        index=True,
+    )
+    move_ids = fields.Many2many(
+        "account.move",
+        "l10n_fr_pdp_flow_move_rel",
+        "flow_id",
+        "move_id",
+        string="Source Moves",
+        help="Invoices that produced this flow.",
+    )
+
+    def _get_default_name(self):
+        """Generate a readable default name."""
+        return _("Flow %(date)s", date=fields.Date.context_today(self))
+
+    @api.model
+    def _selection_flow_type(self):
+        return [
+            (self.TRANSACTION_FLOW_TYPE, _("Flux 10 - Transaction Report")),
+        ]
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        for vals in vals_list:
+            reporting_date = vals.get("reporting_date")
+            period_start = vals.get("period_start")
+            period_end = vals.get("period_end")
+            if reporting_date and not (period_start and period_end):
+                vals.setdefault("period_start", reporting_date)
```

Comment:
i suppose the setdefault are useless there since you are sure that period_start and period_end are not set ?